File: ModernCocoaFarmerBindingsLibrary/Enums/TimePeriod.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Text;

namespace MCF.Time
{
    public enum TimePeriod
    {
        [Description("Day")]
        Day,

        [Description("Month")]
        Month,

        [Description("Year")]
        Year
    }
}

File: ModernCocoaFarmerLibrary/Source/Time/TimePeriod.cpp
#include "Time/TimePeriod.h"
#include "Debug/Assert.h"


namespace CelesteEngine
{
  //------------------------------------------------------------------------------------------------
  template <>
  std::string to_string(MCF::Time::TimePeriod timePeriod)
  {
    using namespace MCF::Time;

    switch (timePeriod)
    {
    case TimePeriod::kDay:
      return "day";
      
    case TimePeriod::kMonth:
      return "month";

    case TimePeriod::kYear:
      return "year";

    default:
      ASSERT_FAIL();
      return "";
    }
  }
}